docs/documentation/automation/question_exec.md

```diff
@@ -6,4 +6,105 @@ sidebar_label: Question Exec
 import useBaseUrl from '@docusaurus/useBaseUrl'; 
 import Highlight from '@theme/Highlight';
 
-# Subtitle {#subtitle}
+<span class="hero__subtitle">
+
+_Customize Survey Questions with Javascript Code_
+
+</span>
+
+## Overview
+Surveys are among Cotalker's main features. Their flexibility and customization ability allow them to adapt to each business's needs.
+
+This spec expands the notion of survey functionality by introducing _Question EXEC_, custom code execution for your survey questions.
+
+:::info
+- Custom code must be written in **Javascript**.
+- Code should always be previously _sandboxed_.
+:::
+
+## Set Up
+
+To add your custom code, choose any survey component and select the **Automation** tab. The code is added in the field at the bottom of the settings panel.
+_See the image below:_
+
+<img alt="automation tab" class="img_sizing_narrow item shadow--md" src={useBaseUrl('img/automation_question_exec_00.png')} />
+<br/>
+<br/>
+
+You can add custom code in any of the four _stages_ of a _survey component's lifecycle_. Set the **stage** either with the dropdown menu or by clicking the corresponding icon. 
+_See the image below:_
+
+<img alt="automation tab" class="img_sizing_narrow item shadow--md" src={useBaseUrl('img/automation_question_exec_01.png')} />
+<br/>
+
+## Component Lifecycle Stages
+- **Preload**: The code executes when a new survey instance is summoned.
+- **onDisplay**: Code executes when a survey is in editing mode.
+- **Validate**: Code executes before internal validations, which permits adding additional levels of validation (business guidelines, protocols, etc.).
+- **Postsave**: Code executes after custom and internal validations.
+
+## Coding Guide
+### Available Commands:
+_The following commands are available for use in your code:_
+- **SET_READONLY**: Sets the field to read-only mode.
+- **SET_RESPONSES**: Replaces the value stored in the field.
+- **SET_REQUIRED**: Sets the field as mandatory, requiring an answer.
+
```

Comment:
We should document, how to use these commands.

NOTE: The QuestionExec function, MUST:
1. Contain a function named run, without any arguments e.g., `const run = () => { ... };` or `  function run () { ... } `  
2. Return an Array of Commands
3. NOTE: A Command has the structure { cmd: <cmd-name>, value: <cmd-value>, result: <cmd-result>  }
4. cmd-name, cmd-value and cmd-result depends on the command.

E.g., SET_READONLY  has value "true" or "false", and no result
We can show this as a table of possible values.

Reply:
Lets go point by point
(1) This is the most general description of Question Exec. The programmer MUST write a function named `run` to make this work:

```
const run = () => { yourCode };
```
```
function run () {  yourCode }
```

Reply:
(2) (3) 
This function -run- MUST return an Array of Commands.

A Command is a Object with 2 or 3 keys.
* cmd
* value
* result

cmd is a name, from the command list
value meaining depends on the command
result meaning depends on the command, but normally its used to block the execution

Reply:
e.g, 
We can write a simple validation to check if a string has an "@"

``` javascript
function run() {
  const email = context['responses#self'][0] ; // get the value from the input
  if (!email.match(/@/)) return [{ cmd: 'RESULT', result: false, value: 'The input must have an @' }];
  return [{ cmd: 'RESULT', result: true }];
}
```